stream_commands.go

```diff
@@ -418,6 +451,42 @@ func (c cmdable) XTrimMinIDApprox(ctx context.Context, key string, minID string,
 	return c.xTrim(ctx, key, "minid", true, minID, limit)
 }
 
+func (c cmdable) xTrimMode(
+	ctx context.Context, key, strategy string,
+	approx bool, threshold interface{}, limit int64,
+	mode string,
+) *IntCmd {
+	args := make([]interface{}, 0, 7)
+	args = append(args, "xtrim", key, strategy)
+	if approx {
+		args = append(args, "~")
+	}
+	args = append(args, threshold)
+	if limit > 0 {
+		args = append(args, "limit", limit)
+	}
+	args = append(args, mode)
+	cmd := NewIntCmd(ctx, args...)
+	_ = c(ctx, cmd)
+	return cmd
+}
+
+func (c cmdable) XTrimMaxLenMode(ctx context.Context, key string, maxLen int64, mode string) *IntCmd {
+	return c.xTrimMode(ctx, key, "maxlen", false, maxLen, 0, mode)
+}
+
+func (c cmdable) XTrimMaxLenApproxMode(ctx context.Context, key string, maxLen, limit int64, mode string) *IntCmd {
+	return c.xTrimMode(ctx, key, "maxlen", true, maxLen, limit, mode)
+}
+
+func (c cmdable) XTrimMinIDMode(ctx context.Context, key string, minID string, mode string) *IntCmd {
+	return c.xTrimMode(ctx, key, "minid", false, minID, 0, mode)
+}
+
+func (c cmdable) XTrimMinIDApproxMode(ctx context.Context, key string, minID string, limit int64, mode string) *IntCmd {
+	return c.xTrimMode(ctx, key, "minid", true, minID, limit, mode)
+}
```

Comment:
Please add a comment to refactor this and expose similar api to the rest of the commands, i.e. `XTrimWithArgs`

Reply:
Also can add it to our tech debt document to be addressed as part of the next major release.